### PR TITLE
Move contrib documentation to right place.

### PR DESCRIPTION
**Pull request by brenthuisman**
* Move PR message instructions to 'Making a PR' section.
* Move coding instructions to 'Write code' section.

### CODE REVIEW

File: doc/contrib/index.rst [hunk 1]
```diff
@@ -73,7 +73,16 @@ request.
...
-See :ref:`contribpr`.
+Some tips when contributing code:
+
+-  Consider the :ref:`contribcodingstyle`.
+-  If you add functionality, please update the documentation
+   accordingly.
+-  If you add functionality, add tests if applicable. This helps make
+   sure Arbor is stable and functionality does what it’s supposed to
+   do.
+-  If you add/change the public C++ API, provide Python wrappings.
```
Comment (thorstenhater) [inline doc/contrib/index.rst:84]: Wording: add/change => 'modify' or 'work on'

File: doc/contrib/index.rst [hunk 2]
```diff
@@ -73,7 +73,16 @@ request.
...
+-  Make sure Arbor compiles and has no new warnings.
```
Comment (thorstenhater) [inline doc/contrib/index.rst:85]: I would add a point here saying something similar to:
- Add your contribution to the change log (file name goes here)

File: doc/contrib/pr.rst [hunk 3]
```diff
@@ -31,6 +42,9 @@ the workflow is as follows:
 5. A PR on Github is automatically tested by our CI bot called Travis.
    You can also run these tests yourself by building them first
    (``make tests``) and then running them (``./bin/*unit*``).
```
Comment (thorstenhater) [inline doc/contrib/pr.rst:44]: Actually travis does more.

**Review by thorstenhater — CHANGES_REQUESTED** [review]

File: doc/contrib/index.rst (hunk 1 above)
Comment (brenthuisman) [reply doc/contrib/index.rst:84]: Done.

File: doc/contrib/index.rst (hunk 2 above)
Comment (brenthuisman) [reply doc/contrib/index.rst:85]: This we don't do (yet), so I'll hold off on this now.

File: doc/contrib/pr.rst (hunk 3 above)
Comment (brenthuisman) [reply doc/contrib/pr.rst:44]: OK!

**Comment by thorstenhater** [comment]
Please reset the submodule state.

**Review by thorstenhater — CHANGES_REQUESTED** [review]
Looks good, but the pybind11 submodule seems to have changed.

**Comment by brenthuisman** [comment]
I hate submodules.

**Review by thorstenhater — APPROVED** [review]
:+1:

**Pull request merged**